did-jwk/src/lib.rs

```diff
@@ -0,0 +1,248 @@
+        let data = match multibase::Base::decode(&multibase::Base::Base64Url, method_specific_id) {
+            Ok(data) => data,
+            Err(_err) => {
+                // TODO: pass through these errors somehow
+                return (
+                    ResolutionMetadata {
+                        error: Some(ERROR_INVALID_DID.to_string()),
+                        content_type: None,
+                        property_set: None,
+                    },
+                    None,
+                    None,
+                );
+            }
+        };
+
+        let jwk = if let Ok(jwk) = serde_json::from_slice(&data) {
+            jwk
+        } else {
+            return (
+                ResolutionMetadata {
+                    error: Some(ERROR_NOT_FOUND.to_string()),
+                    content_type: None,
+                    property_set: None,
+                },
+                None,
+                None,
+            );
+        };
+        let vm_didurl = DIDURL {
+            did: did.to_string(),
+            fragment: Some("0".to_string()),
+            ..Default::default()
+        };
+        let doc = Document {
+            context: Contexts::Many(vec![
+                Context::URI(DEFAULT_CONTEXT.to_string()),
+                Context::URI("https://w3id.org/security/suites/jws-2020/v1".to_string()),
+            ]),
+            id: did.to_string(),
+            verification_method: Some(vec![VerificationMethod::Map(VerificationMethodMap {
+                id: vm_didurl.to_string(),
+                type_: "JsonWebKey2020".to_string(),
+                controller: did.to_string(),
+                public_key_jwk: Some(jwk),
+                ..Default::default()
+            })]),
+            assertion_method: Some(vec![VerificationMethod::DIDURL(vm_didurl.clone())]),
+            authentication: Some(vec![VerificationMethod::DIDURL(vm_didurl.clone())]),
+            capability_invocation: Some(vec![VerificationMethod::DIDURL(vm_didurl.clone())]),
+            capability_delegation: Some(vec![VerificationMethod::DIDURL(vm_didurl.clone())]),
+            key_agreement: Some(vec![VerificationMethod::DIDURL(vm_didurl)]),
+            ..Default::default()
+        };
+        (
+            ResolutionMetadata::default(),
+            Some(doc),
+            Some(DocumentMetadata::default()),
+        )
+    }
+}
+
+impl DIDMethod for DIDJWK {
+    fn name(&self) -> &'static str {
+        "jwk"
+    }
+
+    fn generate(&self, source: &Source) -> Option<String> {
+        let jwk = match source {
+            Source::Key(jwk) => jwk,
+            Source::KeyAndPattern(jwk, pattern) => {
+                if !pattern.is_empty() {
+                    // pattern not supported
+                    return None;
+                }
+                jwk
+            }
+            _ => return None,
+        };
+        let jwk = jwk.to_public();
+        let jwk = serde_jcs::to_string(&jwk).unwrap();
```

Comment:
From [docs.rs](https://docs.rs/serde_jcs/latest/serde_jcs/fn.to_string.html):

> Serialization can fail if T's implementation of Serialize fails.

If `JWK` is safe to serialize then it shouldn't be a problem, otherwise I can add error handling.

Reply:
IMO we should change the `.unwrap()` to `.ok()?`